x-pack/plugins/security_solution/cypress/README.md

```diff
@@ -2,6 +2,50 @@
 
 The `security_solution/cypress` directory contains functional UI tests that execute using [Cypress](https://www.cypress.io/).
 
+Currently with Cypress you can develop `functional` tests and coming soon `CCS` and `Upgrade` functional tests.
+
+If you are still having doubts, questions or queries, please feel free to ping our Cypress champions:
+
+- Functional Tests:
+  - Gloria Hornero (@Glo) and Patryk Kopycinsky (@patryk.kopycinski) 
+  
+- CCS Tests:
+  - Technical questions around the https://github.com/elastic/integration-test repo:
+     - Domenico Andreoli (@dom)
+  - Doubts regarding testing CCS and Cypress best practices:
+     - Gloria Hornero (@Glo)       
+
+## Table of Contents
+
+[**How to add a new Cypress test**](#how-to-add-a-new-cypress-test)
+
+[**Running the tests**](#running-the-tests)
+
+[**Debugging your test**](#debugging-your-test)
+
+[**Folder structure**](#folder-structure)
+
+[**Test data**](#test-data)
+
+[**Development Best Practices**](#development-best-practices)
+
+[**Test Artifacts**](#test-artifacts)
+
+[**Linting**](#linting)
+
+## How to add a new Cypress test
+
+Before adding a new test please ask yourself the following question: `Can I test this behaviour with the unit or API tests?` if the answer is `Yes I can`, 
+then, please add the test there since Cypress is not the best place for it :) 
+
+If the answer is `No, I can't` then this is the best place to start.
```

Comment:
Is this the philosophy we want? Prefer unit tests over e2e tests?

Typically I try to add both, and in the past I have heard or seen contradictions where developers prefer e2e tests instead of unit tests. Which I don't 100% agree with.

I would almost say that since the in-determinism rises with e2e tests but unit tests are more deterministic, we should encourage the typical philosophy of try to add more unit tests than e2e tests as your base code coverage but then add Cypress UI tests to test the basics of the user interface.

You could add some notes that we shouldn't be testing REST API and data contract behaviors (which Cypress is not great at) and instead use the backend e2e functional test runners for that?

Really Cypress's sweet spot is testing that the UI clicking and flushing out of details is working for us.

Reply:
Thanks @FrankHassanabad for the suggestion I'll rephrase that part. What I would like to avoid is the use of Cypress for things that can be checked with just one unit test or having duplications.